internal/transport/keepalive_test.go

```diff
@@ -307,14 +287,8 @@ func (s) TestKeepaliveClientOpenWithUnresponsiveServer(t *testing.T) {
 	}
 	defer conn.Close()
 
-	// Give keepalive some time.
-	time.Sleep(4 * time.Second)
-
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	// Make sure the client transport is healthy.
-	if _, err := client.NewStream(ctx, &CallHdr{}); err != nil {
-		t.Fatalf("client.NewStream() failed: %v", err)
+	if err := pollForHealthyStream(client); err != nil {
```

Comment:
I think in this case we have no choice but to sleep (but for <<<4s), as in the test above that uses `pollForHealthyStream`.